### DISCUSSION TITLE
Smth is going wrong

### DISCUSSION THREAD

**Discussion by danielpodgornii** [mailhexu/TB2J#25] [Q&A]
Hello! I want to run tb2j siesta module from the exemplex, and something went wrong. Please i need your help!

danpdg95@N5:/media/danpdg95/Disk D/TB2J-master/examples/Siesta/BccFe$ python3 siesta2J.py --fdf_fname siesta.fdf --elements Fe --kmesh 7 7 7

TB2J version 0.7.4.3
Copyright (C) 2018-2020  TB2J group.
This software is distributed with the 2-Clause BSD License, without any warranty. For more details, see the LICENSE file delivered with this software.

Traceback (most recent call last):
  File "/media/danpdg95/Disk D/TB2J-master/examples/Siesta/BccFe/siesta2J.py", line 139, in <module>
    run_siesta2J()
  File "/media/danpdg95/Disk D/TB2J-master/examples/Siesta/BccFe/siesta2J.py", line 120, in run_siesta2J
    gen_exchange_siesta(
  File "/home/danpdg95/.local/lib/python3.10/site-packages/TB2J/manager.py", line 269, in gen_exchange_siesta
    if H.spin.is_colinear:
AttributeError: 'NoneType' object has no attribute 'spin'
danpdg95@N5:/media/danpdg95/Disk D/TB2J-master/examples/Siesta/BccFe$

**Comment by mailhexu** [comment]
Hi, 
Thanks for reporting. 
I guess you are using the most recent version of SIESTA? I meet this problem too. It is because sometimes the Hamitonian file is broken randomly in this version. I am tring to fix it. 
I suggest that you install an older version of SIESTA and use that for the moment, e.g. Max-1 version.
https://gitlab.com/siesta-project/siesta/-/wikis/Guide-to-Siesta-versions
Best wishes,
HeXu

**Comment by danielpodgornii** [comment]
Hi!
Thanks for responding.
This mistake came from the exemple from TB2J-master folder. Does older
versions of SIESTA compute SOC calculation?

*Daniel Podgornii*



Laboratorul Metode Fizice de Studiere

a Solidului "T.I. Malinowski"

Institutul de Fizica Aplicata al ASM


În mie., 11 ian. 2023 la 11:57, Xu He ***@***.***> a scris:

**Comment by mailhexu** [comment]
Hi, 
Yes SOC is possible in the older version if DFT+U is not used. 
I will send you a alert if the problem is fixed in the new version of SIESTA. 
HeXu

**Comment by danielpodgornii** [comment]
Thanks you a lot!

чт, 12 янв. 2023 г., 11:57 Xu He ***@***.***>:
